packages/core/src/tools/mcp-client.ts

```diff
@@ -1239,6 +1241,31 @@ export async function createTransport(
   mcpServerConfig: MCPServerConfig,
   debugMode: boolean,
 ): Promise<Transport> {
+  if (mcpServerConfig.authProviderType === 'service_account_impersonation') {
```

Comment:
Suggestion: Consider updating this to use a switch on authProvider type that way you can do an exhaustive check on AuthProviderType values.